src/sources/prometheus/parser.rs

```diff
@@ -15,11 +15,13 @@ fn has_values_or_none(tags: BTreeMap<String, String>) -> Option<BTreeMap<String,
     }
 }
 
-fn utc_timestamp(timestamp: Option<i64>) -> Option<DateTime<Utc>> {
-    timestamp.and_then(|timestamp| {
-        Utc.timestamp_opt(timestamp / 1000, (timestamp % 1000) as u32 * 1000000)
-            .latest()
-    })
+fn utc_timestamp(timestamp: Option<i64>, now: DateTime<Utc>) -> DateTime<Utc> {
```

Comment:
I might call `now` `default` here given how it is used.

Reply:
Will do.